src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -557,6 +573,64 @@ string tryHandleNumericLimits(ref bool unnamedEnumValue)
                 return null;
             }
 
+            string tryHandleTypeUnionLimits()
+            {
+                if (evaluations.IsEmpty && ValueSetFactory.TypeUnionValueSetFactoryForInput(input) is { } factory &&
```

Comment:
To check my understanding, and, in case it helps the second reviewer:

It looks like `evaluations.IsEmpty` ensures that we don't propose one of our unmatched types as an example pattern, in cases where we would move on to find a more suitable example further along in the process, For example, in test `Exhaustiveness_04`, this allows us to propose pattern `false` as an example rather than pattern `bool`.